File: src/main/java/org/digma/intellij/plugin/jaegerui/model/incoming/SpanListPayload.java
package org.digma.intellij.plugin.jaegerui.model.incoming;

import com.fasterxml.jackson.annotation.*;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;

import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
@JsonSerialize
public record SpanListPayload(@JsonProperty(value = "spans",required = true) List<Span> spans) {

    @JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
    public SpanListPayload {
        //nothing to do , for jackson only
    }
}

File: src/main/kotlin/org/digma/intellij/plugin/ui/jaegerui/model/SpanListPayload.kt
package org.digma.intellij.plugin.ui.jaegerui.model

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import java.beans.ConstructorProperties

@JsonIgnoreProperties(ignoreUnknown = true)
data class SpanListPayload
@JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
@ConstructorProperties("spans")
constructor(
    val spans: List<Span>
)